LoRaEngine/LoraKeysManagerFacade/IRegistryPageResult.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    public interface IRegistryPageResult<TResult>
+        where TResult : class
+    {
+        Task<IEnumerable<TResult>> GetNextPageAsync();
```

Comment:
Async operations should take a `CancellationToken`.

Reply:
Also consider returning a `Task<ICollection<TResult>>` to require the sequence of results to be materialised.